### PR TITLE
issue #2569: added model tests for Go

### PR DESCRIPTION
**Pull request by guohuang**
@wing328 i have got the test running, but there are couple errors that i cant figure it out, please advise

image

Link #2569

### CODE REVIEW

File: modules/swagger-codegen/src/test/java/io/swagger/codegen/go/GoClientOptionsTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    @Override
+    protected CodegenConfig getCodegenConfig() {
+        return clientCodegen;
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    protected void setExpectations() {
+        new Expectations(clientCodegen) {{
+/*            clientCodegen.setModelPackage(GoClientOptionsProvider.MODEL_PACKAGE_VALUE);
```
Comment (guohuang) [inline modules/swagger-codegen/src/test/java/io/swagger/codegen/go/GoClientOptionsTest.java]: not sure what this is for, i get additional errors if i enable these lines.

**Comment by guohuang** [comment]
@wing328  i finally get everything running, but i am sure some of the fields are not needed, please advise.

**Comment by guohuang** [comment]
@wing328 code has been updated per your request, all tests passed

**Comment by wing328** [comment]
@guohuang thanks for the quick turnaround. I'll take a look and merge accordingly.

**Comment by wing328** [comment]
@guohuang looks good. I've left some comments for minor enhancements that you can do in another PR.

File: modules/swagger-codegen/src/test/java/io/swagger/codegen/go/GoModelTest.java [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        final DefaultCodegen codegen = new GoClientCodegen();
+        final CodegenModel cm = codegen.fromModel("sample", model);
+
+        Assert.assertEquals(cm.name, "sample");
+        Assert.assertEquals(cm.classname, "Sample");
+        Assert.assertEquals(cm.description, "a map model");
+        Assert.assertEquals(cm.vars.size(), 0);
+        // {{imports}} is not used in template
+        //Assert.assertEquals(cm.imports.size(), 2);
+        //Assert.assertEquals(Sets.intersection(cm.imports, Sets.newHashSet("Children")).size(), 1);
```
Comment (wing328) [inline modules/swagger-codegen/src/test/java/io/swagger/codegen/go/GoModelTest.java]: Unlike PHP, we can test {{imports}} as well, which is used in the Go mustache template.

**Pull request merged**